### PR TITLE
Add more tests for ec2.py

### PR DESCRIPTION
**Pull request by aelsabbahy**
Increase test coverage in preparation for boto2 -> boto3 upgrade

### CODE REVIEW

**Comment by tardyp** [comment]
cool!

File: master/buildbot/test/unit/test_worker_ec2.py [hunk 1]
```diff
@@ -138,7 +138,7 @@ def test_start_vpc_instance(self):
         bs = ec2.EC2LatentWorker('bot1', 'sekrit', 'm1.large',
                                  identifier='publickey',
                                  secret_identifier='privatekey',
-                                 keypair_name="test_key",
+                                 keypair_name="latent_buildbot_slave",
```
Comment (rutsky) [inline master/buildbot/test/unit/test_worker_ec2.py]: Please use `latent_buildbot_worker` instead of `latent_buildbot_slave`.

**Comment by ryansydnor** [comment]
Lgtm! (Other than rutsky's comment)

File: master/buildbot/test/unit/test_worker_ec2.py (hunk 1 above)
Comment (aelsabbahy) [inline master/buildbot/test/unit/test_worker_ec2.py]: The test fixture sets up `latent_buildbot_slave`, I can change the test fixture to use `latent_buildbot_worker` and update the whole test suite , but I would also have to change the defaults in [ec2.py#L76](https://github.com/buildbot/buildbot/blob/c3969c43b861f00abf7653c02b80cd0e8b8741a6/master/buildbot/worker/ec2.py#L76)

Want me to change it for both, or is there some backwards compatibility that needs to be maintained? for `_slave`?

Comment (rutsky) [inline master/buildbot/test/unit/test_worker_ec2.py]: Default can't be changed due to backward compatibility.
Current use of default value is deprecated and warning is emitted, if user doesn't explicitly specify value for `keypair_name` or `security_name`.

Please change value that is used in test.

Comment (aelsabbahy) [inline master/buildbot/test/unit/test_worker_ec2.py]: I changed the value from `test` to `worker` in every test that wasn't checking for deprecation warnings.. there's quite a few that did that.

**Comment by aelsabbahy** [comment]
Let me know if any other changes are required.

**Comment by rutsky** [comment]
LGTM

**Comment by aelsabbahy** [comment]
Once this is merged, I'll submit the boto2 -> boto3 upgrade as a new PR.

**Pull request merged**